observing/makesdf.py

```diff
@@ -1,150 +1,135 @@
 import pandas as pd
 import numpy as np
 from astropy.time import Time
-import warnings
 from datetime import timedelta
-import subprocess
-import argparse
+import random
+import os
+import logging
+
+logger = logging.getLogger('observing')
 
-MODES = ['POWER', 'VOLT', 'FAST', 'SLOW']
 
+MODES = ['POWER', 'VOLT', 'FAST', 'SLOW']
+EPHEM_MODES = ['TRK_JOV', 'TRK_SOL', 'TRK_LUN']
 
-def create(out_name):
+def create(out_name, sess_id=None, sess_mode=None, beam_num=None, cal_dir='/home/pipeline/caltables/latest', pi_id=None,
+           pi_name=None, config_file=None, n_obs=1, obs_mode=None, obs_start=None, obs_dur=None, ra=None, dec=None,
+           obj_name=None, int_time=None):
     """ Create a file out_name as an SDF
     """
+
     sdf_text = ''
 
-    print(" ")
-    print("Enter a session ID and session mode with the format NNN MODE")
-    correct = False
-    while not correct:
-        inp = input()
-        out = inp.split(' ')
-        if len(out) < 2:
-            raise Exception(f"{len(out)} inputs instead of the expected 2 were given")
-        sess_id = out[0]
-        sess_mode = out[1]
-        if sess_mode in MODES:
-            correct = True
-        elif sess_mode not in MODES:
-            print(f"The session mode provided is not recognized. Please give one of the following: {MODES}")
+    if sess_id is None:
+        sess_id = random.randint(0, 1000)
```

Comment:
um......

Delay the validation (and any validations that will require talking to the db) till the very end. Refactor this into a function and use a db transaction to assign a sess_id, cuz I assume you are using sess_id as the primary key?

Reply:
I have another branch that is smarter around here (but not as smart as using a db transaction). I'll avoid making a change here at this point.